### ISSUE TITLE
ars magica levels reset when traveling between dimensions/ ui disappears on death

### ISSUE THREAD

**Issue by Blortias** [Mithion/ArsMagica2#1613]
i recently put a small modpack together for some friends of mine to use, and I've noticed every time one of us dies the Ars Magica ui disappears. We also can't use the occulus. in order for us to use Ars Magica again we need to re log. another issue I've noticed is when traveling between dimensions mainly "the end" Ars Magica levels will reset only when we leave to go back to the overworld. only recently have i noticed it doing the same thing in the nether. when looking into the problem i found a post stating the same things happening but the post was from 2 years ago and the issue was apparently resolved in 1.2 while the version im using is 1.4

**Comment by xenxero1** [comment]
A temporary fix is the code /am2 forcesync  .   I have no idea if that's correct, but I know theres a similar code that force-syncs your client with the servers stored data. Might have to do it a couple times for all the functions to be restored.

**Comment by Blortias** [comment]
ok thanks ill try that and see if it works

**Comment by Blortias** [comment]
after a few days of trying to get this to work i cannot seem to make this work the issue persists

**Comment by TheIcyOne** [comment]
What is the full name of the AM2, forge and Minecraft versions you are running, and are there any other mods alongside these?

**Comment by xenxero1** [comment]
I know it was a big issue on -17b, I even submitted a few related bug reports. I haven't had the nerve to retest it on -18 however.

**Comment by Blortias** [comment]
1.7.10_AM2-1.4.0.008
forge-1.7.10-10.13.4.1614

as for the mods...
AnimationAPI-1.7.10-1.2.4
appliedenergistics2-rv2-stable-10
backpack-2.0.1-1.7.x
BetterStorage-1.7.10-0.12.0.124
BiblioCraft[v1.11.5][MC1.7.10]
Botania r1.8-249
BrandonsCore-1.0.0.12
buildcraft-7.1.17
carpenters blocks v3.3.7 dev - mc 1.7.10
Chisel-Mod-1.7.10
CodeChickenCore-1.7.10-1.0.7.47-universal
CoFHCore-[1.7.10]3.1.3-327
Draconic-Evolution-1.7.10-1.0.2h
EnderIO-1.7.10-2.2.8.381
EnderStorage-1.7.10-1.4.5.29-universal
ExtraTiC-1.7.10-1.4.6
extrautilities-1.2.12
forestry_1.7.10-4.2.15.63
IguanaTinkerTweaks-1.7.10-2.1.6
Industrial-Craft-2-Mod-Experimental-1.7.10
ironchest-1.7.10-6.0.62.742-universal
magicalcrops-4.0.0_PUBLIC_BETA_2
Mantle-1.7.10-0.3.2b
MineFactoryReloaded-[1.7.10]2.8.1-174
NotEnoughItems-1.7.10-1.0.5.120-universal
openblocks-1.7.10-1.5.1
OpenModsLib-1.7.10-0.9.1
OptiFine_1.7.10_HD_U_D6
Pams HarvestCraft 1.7.10h
PlayerAPI-1.7.10-1.4
Reis-Minimap-Mod-1.7.10
runicdungeons-1.1.3
Secret-Rooms-Mod-1.7.10 (1)
SmartMoving-1.7.10-15.6
SmartRender-1.7.10-2.1
TConstruct-1.7.10-1.8.8
Thaumcraft-1.7.10-4.2.3.5
ThermalDynamics-[1.7.10]1.2.0-171
ThermalExpansion-[1.7.10]4.1.4-247
ThermalFoundation-[1.7.10]1.2.5-115
twilightforest-1.7.10-2.3.7
VeinMiner-0.34.1-1.7.10r473+4d066f1
Waila-1.5.10_1.7.10
WailaHarvestability-mc1.7.x-1.1.2

That's all the mods i have on right now. i really don't understand this issue mainly because all these mods are present in the Crackpack as well. Iv'e had no issue with Ars Magica on the Crackpack.(also looking through my mods for this iv'e realized i forgot reliquary...)

**Comment by TheIcyOne** [comment]
Try updating to AM2 *.009

**Comment by Blortias** [comment]
there has still been no changes the problem persists

**Comment by TheIcyOne** [comment]
Do any of these mod versions differ from those present in your Crackpack instances?

**Comment by Blortias** [comment]
no not at all I grabbed them all from the crackpack instance 
thats why i dont understand this issue

**Comment by Blortias** [comment]
i dont know what else to do
and nobody is rly playing on my server anymore so...

**Comment by xenxero1** [comment]
Move to reopen? This is still a glitch that needs to be addressed. You can turn off notifications for the issue if you don't want to be spammed, but there is valid info here that needs to be addressed and referenced. Thank you.

**Comment by TheIcyOne** [comment]
Not sure a fix will occur for 1.7 versions, since we've been having a lot of difficulty reproducing it.

**Comment by xenxero1** [comment]
I've been able to openly reproduce it in -17b and -18. Using Spinycraft V2 1.2.1 modpack.

**Comment by TheIcyOne** [comment]
Could you list the steps you take to reproduce it, I've been having no luck on my end.

**Comment by xenxero1** [comment]
For Smoke-testing sake I believe this should trigger it:

1. Create a new world
2. Begin Ars Magica
3.  Attain level 5
4. (Optional) Have 2 soulbound items in your inventory
5. Enter The End
6. Fight/kill the dragon, Spend a couple minutes exploring the outer edges. (At this point you might experience the Am2 Level/UI Desync)
7. Return to the Overworld via Portal.
8. Observer any changes in level, Am2 Access permissions (occulus, ArCompendium)
9. Run /am2 forcesync (or whatever command it is for forcesyncing)
End Test.

I haven't been able to reproduce it with the nether, or custom dimensions from Just Enough Dimensions (Transporting via Command Block teleports). Untested with anything like Aether or RFtoolsDims. I've got a feeling one of those might trigger it, or atleast like Mysticraft.

**Comment by xenxero1** [comment]
Honestly for me it occurs anytime me or another player enters the End. Hasn't been a huge issue, but it should be noted i've only ever reproduced it on SMP

**Comment by TheIcyOne** [comment]
So the desync occurs even without a death?

**Comment by TheIcyOne** [comment]
Other than the one caused by hacky end portals of hackiness.

**Comment by xenxero1** [comment]
Correct. I can't recall any instances of it happening on death although i can't confirm for sure. Just the Pseudo-death from the end portal. Which also seems to cause issues with soulbound items, but thats a seperate issue.

**Comment by morphtrust** [comment]
I am having this same issue with my players too, if you need to try using my server to replicate and want server logs I can get you that info, as well as my modpack that they are using which is at
https://www.technicpack.net/modpack/okmcmdpk.1109971 

note this is an updated version of another modpack I have on that site to see all the mods I use in it, go to my profile and look at the original one (giantkin modpack something or other for skywaycraft) as it and the giant robot versions are the only other two modpacks I have made. eventually the first one I listed will be merged with the first one with older mods (thats pretty much all I did was update the mods all around as much as I could since I could not get forge 15xx of any sort to run, as a server in the versions of cauldron I have been able to find.) if you want to reach me you can find me on discord in the group 
https://discord.gg/dxKuMD4  called OKcash and there is a channel on the group called #minecraft-og  that I respond pretty quickly to if I am online, and at my computer. 

I should add the minecraft server we have it all running on and are having this issue is in the pinned messages too, just let me know and I will switch you to builder status and send you logs when you get done testing if you need. as well as any other help I can offer.

**Comment by Blortias** [comment]
just now getting notifications for this in my email sorry it took so long
i also remember now how dumb i was giving you a literal list when i could give you the link i put up to download my mod pack. ill catch that really quick
if you get the ars magica book get a few levels then go into the end and die your ui will disappear and on relogging i will reapear (if i remember correctly) but you will be starting at level 0 although all your unlocks in the occulus will still be there



its a media fire link but i think it has the older version of ars magica and not .009
http://www.mediafire.com/file/kujuk11iskeh0gx/Blortcraft.zip

**Comment by Blortias** [comment]
right reading back over this
on my mod pack any exit from the end even teleporting out resets your ars magica levels
just dieing will be quicker unless you want to spend the time killing the ender dragon

**Comment by TheIcyOne** [comment]
To be perfectly honest, you probably won't get a fix for this, the version has been unsupported for a ridiculously long time and the mod itself is being totally reworked. This doesn't seem to be reproducible from your description in a pure AM2 setting, so you'll have to try to find the culprit mod(s) yourself.

**Comment by Blortias** [comment]
well im sure the rework is for what 1.10 and up (my pc atm cant even handle a few small mods from 1.10 for some reason) so i would like to leave this open for anyone else who has this issue so if a fix ever does come out they can find it here

**Comment by TheIcyOne** [comment]
The rework is for 1.12+, IIRC.

**Comment by Blortias** [comment]
its obviously a rare bug so idk how it really happens idk the in depth coding problems or compatability sorta stuff but it quite a hassle so i would like it to get resolved i mean cus i know not everyone likes jumping to the laster version of mc and using those mods. so for anyone who does happen to have this issue i would like there to be a solution for it. and i mean im not done with my modpack ive got other things to fix and balance  so just in case ill leave this open.

**Comment by jivvy** [comment]
Relog. Fixed for me
